test/fw/ptl/lib/pbs_testlib.py

```diff
@@ -8471,6 +8471,9 @@ def cleanup_jobs(self):
             select_xt = 'x'
         jobs = self.status(JOB, extend=select_xt)
         job_ids = sorted(list(set([x['id'] for x in jobs])))
+        running_jobs = self.filter(JOB, {'job_state': 'R'})
```

Comment:
You don't need to call Server.filter here (which internally will run qstat again), you already have the full status of jobs in `jobs`, you can make use of that then filter out running jobs like:
```python
last_running_job = job_ids[-1]
running_jobs = sorted([x['id'] for x in jobs if x['job_status'] == 'R'])
if running_jobs:
    last_running_job = running_jobs[-1]
```

Reply:
Won't, sorted() do an alpha sort?  Don't we need to chop the IDs at the first '.' so sorted() does a numeric sort?  So:
```
running_jobs = sorted(x['id'].split('.')[0] for x in jobs if x['job_state'] == 'R')
```

Reply:
we need to make it an int() as well right? I think we should do sorted(int(x['id'].split('.')[0]) ...

Reply:
I have not sorted the list of jobs, I do not think we will need it since the status of jobs will return us jobs in the sequence they are present on the server.
Do correct me if I am wrong

Reply:
Actually don't care about sorting persay.  We care about the running job we had on the qdel line.  This means we should sort both lists or sort neither.